pandas/tests/plotting/test_frame.py

```diff
@@ -166,7 +166,7 @@ def test_integer_array_plot(self):
 
     def test_mpl2_color_cycle_str(self):
         # GH 15516
-        colors = ['C' + str(x) for x in range(10)]
+        colors = ['c' + str(x) for x in range(10)]
```

Comment:
Wait, is this really what raises the warning? `C0`, etc. is correct, and `c0` should not work (assuming these values are directly passed to Matplotlib.) Is Pandas doing some other processing of these colour names?

Reply:
OK, looking at `_get_standard_colors`, I see Pandas does do extra changes, it checks the individual elements of both `[colors]` and `list(colors)`. The latter changes `'C0'` to `['C', '0']`, neither of which are valid.

I'm pretty sure this test was correct and it is in _that_ function that a warning capture context should be added.

Reply:
@jorisvandenbossche any thoughts on this?  I think you introduced this test a while back.

Reply:
Reverted this change, can address separately.

Reply:
Didn't check in detail, but as @QuLogic says, this test is certainly testing something valuable, that `'C0'`, `'C1'`, etc color names are working, and should not be changed like the above.